File: java/example/src/test/java/vanilladi/BarSpy.java
package vanilladi;

import java.util.ArrayList;
import java.util.List;

public class BarSpy extends Bar {
    private List<Integer> calledArgs = new ArrayList<>();

    public Iterable<Integer> getCalledArgs() {
        return calledArgs;
    }

    @Override
    public void doSomething(int value) {
        calledArgs.add(value);
    }
}

File: kotlin/example/src/test/kotlin/vanilladi/BarSpy.kt
package vanilladi;

class BarSpy : Bar() {
    private val calledArgs = mutableListOf<Int>();

    fun getCalledArgs(): Iterable<Int> {
        return calledArgs;
    }

    @Override
    override fun doSomething(value: Int) {
        calledArgs.add(value);
    }
}
